runtime/lua/vim/treesitter/dev.lua

```diff
@@ -436,11 +459,145 @@ function M.inspect_tree(opts)
     buffer = buf,
     once = true,
     callback = function()
-      if api.nvim_win_is_valid(w) then
-        api.nvim_win_close(w, true)
+      close_win(w)
+    end,
+  })
+end
+
+local preview_ns = api.nvim_create_namespace('treesitter/dev-preview')
+
+---@param query_win integer
+---@param buf integer
+local function update_preview_highlights(query_win, buf)
+  local parser = vim.treesitter.get_parser(buf)
+  local lang = parser:lang()
+  local query_buf = api.nvim_win_get_buf(query_win)
+  api.nvim_buf_clear_namespace(buf, preview_ns, 0, -1)
+  local topline, botline = vim.fn.line('w0', query_win), vim.fn.line('w$', query_win)
+  local query_content =
+    table.concat(api.nvim_buf_get_lines(query_buf, topline, botline, false), '\n')
+
+  local ok_query, query = pcall(vim.treesitter.query.parse, lang, query_content)
+  if not ok_query then
+    return
+  end
+
+  local cursor_word = vim.fn.expand('<cword>') --[[@as string]]
+  -- Only highlight captures if the cursor is on a capture name
+  if cursor_word:find('^@') == nil then
+    return
+  end
+  -- Remove the '@' from the cursor word
+  cursor_word = cursor_word:sub(2)
+  for id, node in query:iter_captures(parser:trees()[1]:root(), buf, 0, -1) do
```

Comment:
I thought you said you limited the query range. Did it not work?

Reply:
Ah, I think you misunderstood me. I meant the highlights in the main buffer. We don't need to run the query over the whole buffer.

Reply:
No it's okay, I understood what you meant but my brain melted for a second. Fixing it.